### ISSUE TITLE
mini cart does not clear after placing an order - magento 2.0.7

### ISSUE THREAD

**Issue by nmallepally** [magento/magento2#5621]
Hello,

After i have completed the purchase or placing an order the cart still shows the items and does not remove itself untill any new products added again. it is misleading. can any one give some insight on this?. do i need to upgrade 2.1 to fix or is there any alternative solution.

Thanks,
Navya

**Comment by andimov** [comment]
@nmallepally 
Thanks for reporting. 
Please, format this issue according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines): with steps to reproduce, actual result and expected result.

**Comment by nmallepally** [comment]
Hi @andimov 
Updated 2.0.7 to 2.1
Product added n checkout
Payment method: Paypal/Cash on Delivery
Payment done successfully
mini cart is not cleared, shows up same products in mini cart.

**Comment by mrkhoa99** [comment]
@nmallepally you can read more here: http://magento.stackexchange.com/questions/127203/magento-2-minicart-does-not-clear-items-after-checkout/127347#127347

**Comment by andimov** [comment]
@nmallepally
Did you check solution from @mrkhoa99 ?

**Comment by nmallepally** [comment]
@andimov I'm checking it after the changes, will update in few minutes.

**Comment by nmallepally** [comment]
@andimov It returned me a different errors.
Applied changes from this [fix ](https://github.com/magento/magento2/pull/5807/files#diff-1)
After payment the page returns a network error, page not working on success page, when clicked on reload.
Returns a page checkout/cart saying not items in the cart, which it should not and yet displays the items in the cart.
During the above process payment is made and order is generated.
Attaching images for reference:
![cart error](https://cloud.githubusercontent.com/assets/19605497/17474267/029f70f6-5d4d-11e6-8bb9-58679b5e90f3.png)
![error cart 1](https://cloud.githubusercontent.com/assets/19605497/17474310/2ec445ee-5d4d-11e6-8147-02aa398597c7.png)

**Comment by nmallepally** [comment]
@mrkhoa99 Did you see my result above for the fix?, can you advise any alternative solution as we planning to go live in 2days.

**Comment by mrkhoa99** [comment]
@nmallepally try to get the cart info from [your local storage](http://magento.stackexchange.com/questions/123879/getting-the-product-sku-in-the-header-mini-cart/123903#123903): `localStorage.getItem('mage-cache-storage')`. Seem the local storage not clear. There is a temporary solution for you, navigate to `vendor/magento/module-checkout/view/frontend/templates/cart/noItems.phtml`, add the following js code:

```
<script>
    require([
        'Magento_Customer/js/customer-data'
    ], function (customerData) {
        var sections = ['cart'];
        customerData.invalidate(sections);
        customerData.reload(sections, true);
    });
</script>
```

**Comment by nmallepally** [comment]
@mrkhoa99 Hi thanks for the quick response. works like a charm for paypal orders, does not clear cart when payment is done for cash on delivery method.

**Comment by mrkhoa99** [comment]
@nmallepally Can you list the files which you applied the paths?

**Comment by nmallepally** [comment]
I'm using magento 2.1v. I just applied the above java script into  vendor/magento/module-checkout/view/frontend/templates/cart/noItems.phtml. As you said this is a temporary solution. Did i miss anything?

**Comment by PieterCappelle** [comment]
Any update on this one? Still exists in 2.1.2.

**Comment by bh-ref** [comment]
encountered this bug in CE 2.1.4 as well. it happens when I **DON'T** sign into my customer account at checkout.

**Comment by refaelgold** [comment]
same here. waiting for update

**Comment by danielmartinezd** [comment]
In EE 2.1.5 Still happening

**Comment by IlnitskiyArtem** [comment]
@nmallepally Thanks for the feedback. Is your issue relevant for latest Magento versions?
We need more information about your configuration to reproduce this issue. Please identify the web server you are running, the versions of PHP and MySQL, and any other information needed to reproduce your issue.

**Comment by nmallepally** [comment]
Hi,
I have updated to Magento version 2.1.6, I have no issues now.
the cart clears after placing an order.

Thanks.

**Comment by silici0** [comment]
Hello,
Im using Magento 2.1.7 fresh install and the problem persist, i have changed /vender/magento, is this the best practice?

**Comment by korostii** [comment]
No, editing `/vendor/magento` isn't considered best practice because your changes will be overwritten with stock files once you update to a newer version of Magento. You'll need to keep track of the all the patches you've applied and re-apply them after each update if you select this method.

**Comment by orlangur** [comment]
I've even met some composer plugins which allow to re-apply patches automatically 🤣 Please don't go this way.

@silici0 just customize needed template according to http://devdocs.magento.com/guides/v2.0/frontend-dev-guide/templates/template-override.html#override mentioning in comment where the patch came from and for which version so that you can easily drop it when it is not relevant anymore due to Magento upgrade.

**Comment by silici0** [comment]
@orlangur Ty, i have uploaded a module that [fix](https://github.com/silici0/MiniCartClear) the problem, im still learning, i would appreciate a code review :)

**Comment by magento-engcom-team** [comment]
@nmallepally, thank you for your report.
The issue is already fixed in develop branch, 2.2.0, 2.1.9

**Comment by mekiran16** [comment]
Hello,
Please add below small script in following file path. it's work for me.

**app/design/frontend/<Vendor>/<theme>/Magento_Checkout/templates/cart/noItems.phtml
app/design/frontend/<Vendor>/<theme>/Magento_Checkout/templates/success.phtml**

```
<script>
require([
     'Magento_Customer/js/customer-data'
 ], function (customerData) {
     var sections = ['cart'];
     customerData.invalidate(sections);
     //customerData.reload(sections, true);
 });
</script>
```

Thanks
